lib/datadog/grok/src/grok.rs

```diff
@@ -64,7 +64,7 @@ impl<'a> Iterator for MatchesIter<'a> {
         // that index, if anything.
         self.names.next().map(|(k, v)| {
             let key = k.as_str();
-            let value = self.captures.at(*v as usize).unwrap_or("");
+            let value = self.captures.at(*v).unwrap_or("");
```

Comment:
There are a lot of these conversions being dropped in this PR, but I don't see the variable types as having changed nor do I recall something in the Rust release notes that referenced it. What has enabled this change?

Reply:
These were all unnecessary casts (i.e. `usize as usize`) that clippy now seems to detect.